### PR TITLE
[REF] *: replace tour_compiler anchor from jQuery to native

### PR DESCRIPTION
**Pull request by pipu-odoo**
In this commit, $anchor jQuery element in tour_compiler (used in run function in stepTour) is replaced by anchor native element.

task~3600484

https://github.com/odoo/enterprise/pull/57493

### CODE REVIEW

File: addons/auth_totp/static/tests/totp_flow.js [hunk 1]
```diff
@@ -54,9 +55,9 @@ function closeProfileDialog({content, totp_state}) {
         content,
         trigger,
         run() {
-            const $modal = this.$anchor.parents('.o_dialog');
-            if ($modal.length) {
-                $modal.find('button[name=preference_cancel]').click()
+            const modal = document.querySelector(".o_dialog");
+            if (modal instanceof HTMLElement) {
```
[the remark this answers is not in the document]
Comment (pipu-odoo) [reply addons/auth_totp/static/tests/totp_flow.js:59]: ok

File: addons/auth_totp/static/tests/totp_flow.js [hunk 2]
```diff
@@ -95,15 +96,19 @@ registry.category("web_tour.tours").add('totp_tour_setup', {
...
     trigger: 'a:contains("Cannot scan it?")',
     async run(helpers) {
-        const $secret = this.$anchor.closest('div').find('[name=secret] span:first-child');
-        const $copyBtn = $secret.find('button');
-        $copyBtn.remove();
+        const secret = this.anchor
+            .closest("div")
+            .querySelector("[name=secret] span:first-child");
+        const copyBtn = secret.querySelector("button");
+        if (copyBtn instanceof HTMLElement) {
```
[the remark this answers is not in the document]
Comment (pipu-odoo) [reply addons/auth_totp/static/tests/totp_flow.js:103]: you don't like instanceof ? :D

File: addons/sale_timesheet/static/tests/tours/sale_timesheet_tour.js [hunk 3]
```diff
@@ -123,17 +124,17 @@ registry.category("web_tour.tours").add('sale_timesheet_tour', {
...
-        const tr = $header[0];
+        const tr = [...header][0];
         let index = -1;
         for (let i = 0; i < tr.children.length; i++) {
             const th = tr.children.item(i);
             if (th.dataset && th.dataset.name === 'qty_delivered')
                 index = i;
         }
-        const qtyDelivered = this.$anchor.find(`tbody > tr:first-child > td.o_data_cell:eq(${index})`).text();
+        const qtyDelivered = queryOne(`tbody > tr:first-child > td.o_data_cell:eq(${index})`, { root: this.anchor }).textContent;
```
[the remark this answers is not in the document]
Comment (pipu-odoo) [reply addons/sale_timesheet/static/tests/tours/sale_timesheet_tour.js:137]: Too much simple ...

**Review by seb-odoo — APPROVED** [review]

**Comment by bastien-ipb** [comment]
@robodoo r+

**Comment by aab-odoo** [comment]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo r+ p=0

**Pull request closed without merging**